lighthouse-core/computed/metrics/responsiveness.js

```diff
@@ -42,20 +67,86 @@ class Responsiveness {
     return responsivenessEvents[index];
   }
 
+  /**
+   * Finds the interaction event that was probably the responsivenessEvent.maxDuration
+   * source.
+   * Note that (presumably due to rounding to ms), the interaction duration may not
+   * be the same value as `maxDuration`, just the closest value. Function will throw
+   * if the closest match is off by more than 4ms.
+   * TODO: this doesn't try to match inputs to interactions and break ties if more than
+   * one interaction had this duration by returning the first found.
+   * @param {ResponsivenessEvent} responsivenessEvent
+   * @param {LH.Trace} trace
+   * @return {EventTimingEvent}
+   */
+  static findInteractionEvent(responsivenessEvent, {traceEvents}) {
+    const candidates = traceEvents.filter(/** @return {evt is EventTimingEvent} */ evt => {
+      // Examine only beginning/instant EventTiming events.
+      return evt.name === 'EventTiming' && evt.ph !== 'e';
+    });
+
+    if (candidates.length) {
+      if (!candidates[0].args.data.frame) {
+        // Full EventTiming data added in https://crrev.com/c/3632661
+        throw new LHError(
+          LHError.errors.UNSUPPORTED_OLD_CHROME,
+          {featureName: 'detailed EventTiming trace events'}
+        );
+      }
+    }
+
+    const {maxDuration, interactionType} = responsivenessEvent.args.data;
+    let bestMatchEvent;
+    let minDurationDiff = Number.POSITIVE_INFINITY;
+    for (const candidate of candidates) {
+      // Must be from same frame.
+      if (candidate.args.data.frame !== responsivenessEvent.args.frame) continue;
+
+      // TODO(bckenny): must be in same navigation as well.
+
+      const {type, duration} = candidate.args.data;
+      // Discard if type is incompatible with responsiveness interactionType.
+      const matchingTypes = interactionTypeToType[interactionType];
+      if (!matchingTypes) {
+        throw new Error(`unexpected responsiveness interactionType '${interactionType}'`);
+      }
+      if (!matchingTypes.has(type)) continue;
+
+      const durationDiff = Math.abs(duration - maxDuration);
+      if (durationDiff < minDurationDiff) {
+        bestMatchEvent = candidate;
+        minDurationDiff = durationDiff;
+      }
+    }
+
+    if (!bestMatchEvent) {
+      throw new Error(`no interaction event found for responsiveness type '${interactionType}'`);
+    }
+    // TODO: seems to regularly happen up to 3ms and as high as 4.
+    if (minDurationDiff > 5) {
+      throw new Error(`no interaction event found within 5ms of responsiveness maxDuration (max: ${maxDuration}, closest ${bestMatchEvent.args.data.duration})`); // eslint-disable-line max-len
```

Comment:
this is 5, but above in jsdoc "4ms" is mentioned as the threshold

Reply:
> this is 5, but above in jsdoc "4ms" is mentioned as the threshold

yeah, it was to be really extra sure but I'll clarify, thx